core/arch/arm/mm/mobj.c

```diff
@@ -159,7 +160,22 @@ struct mobj *mobj_phys_alloc(paddr_t pa, size_t size, uint32_t cattr,
 		return NULL;
 	}
 
-	va = phys_to_virt(pa, battr);
+	switch(battr) {
+	case CORE_MEM_TEE_RAM:
+		area_type = MEM_AREA_TEE_RAM;
+		break;
+	case CORE_MEM_TA_RAM:
+		area_type = MEM_AREA_TA_RAM;
+		break;
+	case CORE_MEM_NSEC_SHM:
+		area_type = MEM_AREA_NSEC_SHM;
+		break;
+	default:
+		DMSG("can't allocate only requested area");
```

Comment:
ok. i'll replace with "can't allocate with specified attribute".

By the way, i added the `CORE_MEM_TEE_RAM` even if it is not yet used. You're fine with that or should it be added only once we start using mobjs in the 'core' memory ?